src/mono/wasm/debugger/BrowserDebugProxy/MemberReferenceResolver.cs

```diff
@@ -372,135 +372,152 @@ public async Task<JObject> Resolve(ElementAccessExpressionSyntax elementAccess,
                     rootObject = indexObject;
                     indexObject = null;
                 }
-                if (rootObject != null)
+                if (rootObject is null || (indexObject is null && elementAccess.ArgumentList is null))
```

Comment:
It's unclear why the second part of condition is relevant. It becomes much clearer if you extract the index string computation to a local method, like:

```csharp
                if (rootObject is null)
                    return null;

                string elementIdxStr = await GetElementIndexString();
                if (elementIdxStr is null)
                    return null;

...


            async Task<string> GetElementIndexString()
            {
                if (indexObject is not null)
                    return indexObject["value"].ToString();

                if (elementAccess.ArgumentList is null)
                    return null;

                string elementIdxStr = string.Empty;
                for (int i = 0 ; i < elementAccess.ArgumentList.Arguments.Count; i++)
                {
                    var arg = elementAccess.ArgumentList.Arguments[i];
                    if (i != 0)
                    {
                        elementIdxStr += ", ";
                        multiDimensionalArray = true;
                    }
                    // e.g. x[1]
                    if (arg.Expression is LiteralExpressionSyntax)
                    {
                        var argParm = arg.Expression as LiteralExpressionSyntax;
                        elementIdxStr += argParm.ToString();
                    }

                    // e.g. x[a] or x[a.b]
                    else if (arg.Expression is IdentifierNameSyntax)
                    {
                        var argParm = arg.Expression as IdentifierNameSyntax;

                        // x[a.b]
                        memberAccessValues.TryGetValue(argParm.Identifier.Text, out var argIndexObject);

                        // x[a]
                        argIndexObject ??= await Resolve(argParm.Identifier.Text, token);
                        elementIdxStr += argIndexObject["value"].ToString();
                    }
                    // indexing with expressions, e.g. x[a + 1]
                    else
                    {
                        string expression = arg.ToString();
                        var argIndexObject = await ExpressionEvaluator.EvaluateSimpleExpression(this, expression, expression, variableDefinitions, logger, token);
                        string type = argIndexObject["type"].Value<string>();
                        if (type != "number")
                            throw new InvalidOperationException($"Cannot index with an object of type '{type}'");
                        elementIdxStr += argIndexObject["value"].ToString();
                    }
                }

                return elementIdxStr;
            }
```

Reply:
Thank you! I preferred not to add refactoring to this PR but if it's hard to read then we can do it now. I just made some small editions, e.g. we cannot return string but string + bool + LiteralExpressionSyntax that I closed in a new `ElementIndexInfo` entity. 
1st `if` that you are changing in the snippet: we cannot just change the logic there, when `rootObject` is null then we are dealing with jagged array (arr[1][1]) and we need to continue resolving using the `indexedObject`, as in fact it's the "new root" now. If we just return there we fail the tests.